lib/improver/blending/weights.py

```diff
@@ -238,12 +242,575 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+    a configuration dictionary.
+    """
+
+    def __init__(self, weighting_coord_name, config_coord_name,
+                 config_dict, weights_coord_name="weights"):
+        """Set up for calculating linear weights from a configuration
+        dictionary.
+
+        Args:
+            weighting_coord_name (str):
+                Name of the coordinate along which the weights will be
+                calculated. For example, if the intention is to provide
+                weights along the forecast_period coordinate within the
+                configuration dictionary, which can be linearly interpolated
+                to create all required weights, then this argument would be
+                "forecast_period".
+                This must be included within the configuration dictionary.
+            config_coord_name (str):
+                Name of the coordinate used for configuration. For example,
+                if the intention is to create weights for different models,
+                then model_configuration may appropriate.
+            config_dict (dict):
+                Dictionary containing the configuration information, namely
+                an initial set of weights and information regarding the
+                points along the specified coordinate at which the weights are
+                valid. An example dictionary is shown below.
+
+        Keyword Args:
+            weights_coord_name (str):
+                Name of the string within the configuration dictionary used
+                to specify the weights. This name is also used as the name
+                of the cube output from this plugin.
+
+        Dictionary of format:
+        ::
+            {
+                "uk_det": {
+                    "forecast_period": [7, 12],
+                    "weights": [1, 0],
+                    "units": "hours"
+                }
+                "uk_ens": {
+                    "forecast_period": [7, 12, 48, 54]
+                    "weights": [0, 1, 1, 0]
+                    "units": "hours"
+                }
+            }
+
+        """
+        self.config_dict = config_dict
+        self.weighting_coord_name = weighting_coord_name
+        self.config_coord_name = config_coord_name
+        self.weights_coord_name = weights_coord_name
+        self._check_config_dict()
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        msg = ("<ChooseWeightsLinearFromDict config_dict = {}, "
+               "weighting_coord_name = {}, config_coord_name = {}, "
+               "weights_coord_name = {}>".format(
+                   str(self.config_dict), self.weighting_coord_name,
+                   self.config_coord_name, self.weights_coord_name))
+        return msg
+
+    def _check_config_dict(self):
+        """Check whether the items within the configuration dictionary
+        are of matching lengths.
+
+        Raises:
```

Comment:
Can you also specify this raises a KeyError if the required items are not present in the dictionary (see `__init__` docstring)?